sdk/src/main/java/com/mapbox/maps/MapView.kt

```diff
@@ -296,9 +300,45 @@ open class MapView : FrameLayout, MapPluginProviderDelegate, MapControllable {
    */
   @SuppressLint("ClickableViewAccessibility")
   override fun onTouchEvent(event: MotionEvent): Boolean {
+    interceptedActionList.firstOrNull()?.let {
+      try {
+        return mapController.onTouchEvent(it)
+      } finally {
+        it.recycle()
+        interceptedActionList.removeFirst()
+      }
+    }
     return mapController.onTouchEvent(event)
   }
 
+  override fun onInterceptTouchEvent(event: MotionEvent): Boolean {
+    return when (event.actionMasked) {
+      MotionEvent.ACTION_DOWN -> {
+        interceptedActionList.add(MotionEvent.obtain(event))
+        false
+      }
+      MotionEvent.ACTION_POINTER_DOWN -> {
+        interceptedActionList.add(MotionEvent.obtain(event))
+        false
+      }
+      MotionEvent.ACTION_MOVE -> {
+        interceptedActionList.any { it.hypot(event) > touchSlop }
+      }
+      else -> {
+        // In general, we don't want to intercept touch events. They should be
+        // handled by the child view.
+        false
+      }
+    }
+  }
+
+  private fun MotionEvent.hypot(event: MotionEvent): Int {
+    return hypot(
+      x = (this?.x ?: event.x) - event.x,
```

Comment:
nit : since `this` is not nullable - can be simplified to `x = this.x - event.x,`